benches/read_many_structs.rs

```diff
@@ -0,0 +1,120 @@
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use ion_rs::lazy::decoder::LazyDecoder;
+use ion_rs::lazy::encoding::TextEncoding_1_1;
+use ion_rs::lazy::r#struct::LazyStruct;
+use ion_rs::lazy::reader::{LazyApplicationReader, LazyTextReader_1_0, LazyTextReader_1_1};
+use ion_rs::lazy::value::LazyValue;
+use ion_rs::lazy::value_ref::ValueRef;
+use ion_rs::IonResult;
+use nom::AsBytes;
+
+pub fn criterion_benchmark(c: &mut Criterion) {
+    const NUM_VALUES: usize = 10_000;
+    let data_1_0 = concat!("{",
```

Comment:
🗺️ Slightly awkward syntax here to produce a compact struct instead of a pretty-printed one.

Reply:
If you feel like it, you could probably use something like
```rust
r"
{
  foo:1,
  bar:2,
}
".split("\n".into()).map(|it| it.trim()).collect::<Vec<_>>().join("");
```

It's still kind of awkward, but it moves the awkwardness somewhere else. I don't have a preference.

Reply:
I'm adding another commit that writes out the struct in full (as you've done) and then just roundtrips it to a `String` of compact text.

The same commit also makes the `IonEq` trait externally visible so the benchmark can confirm that the 1.0 and 1.1 test data are equivalent before measurement begins.